src/backgroundtasks.cs

```diff
@@ -0,0 +1,102 @@
+//
+// BackgroundTasks C# bindings
+//
+// Authors:
+//	Manuel de la Pena Saenz <mandel@microsoft.com>
+//
+// Copyright 2019 Microsoft Corporation All rights reserved.
+//
+using System;
+
+using CoreFoundation;
+using Foundation;
+using ObjCRuntime;
+
+namespace BackgroundTasks {
+
+	[Abstract]
+	[TV (13,0), NoWatch, NoMac, iOS (13,0)]
+	[BaseType (typeof (BGTaskRequest))]
+	interface BGAppRefreshTaskRequest {
+		[Export ("initWithIdentifier:")]
+		IntPtr Constructor (string identifier);
+	}
+
+	[TV (13,0), NoWatch, NoMac, iOS (13,0)]
+	[BaseType (typeof (BGTaskRequest))]
+	interface BGProcessingTaskRequest {
+		[Export ("initWithIdentifier:")]
+		IntPtr Constructor (string identifier);
+
+		[Export ("requiresNetworkConnectivity")]
+		bool RequiresNetworkConnectivity { get; set; }
+
+		[Export ("requiresExternalPower")]
+		bool RequiresExternalPower { get; set; }
+	}
+
+	[Abstract]
+	[TV (13,0), NoWatch, NoMac, iOS (13,0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface BGTaskRequest : NSCopying {
+		[Export ("identifier")]
+		string Identifier { get; }
+
+		[NullAllowed, Export ("earliestBeginDate", ArgumentSemantic.Copy)]
+		NSDate EarliestBeginDate { get; set; }
+	}
+
+	[Abstract]
```

Comment:
I saw the discussion whether we should use `[Abstract]` or not in #macios-community, and one thing that would help decide is tests :smile:

Reply:
Added. Abstract is the way to go, users will have to downcast to the correct Task subclass. The task is created via a request, specified by the user. The test shows how to do it.